AStarAI/src/main/java/com/github/sef24sp4/astarai/caching/DumbCache.java

```diff
@@ -0,0 +1,80 @@
+package com.github.sef24sp4.astarai.caching;
+
+import com.github.sef24sp4.astarai.AStar;
+import com.github.sef24sp4.astarai.Node;
+import com.github.sef24sp4.common.ai.map.Map;
+import com.github.sef24sp4.common.entities.ICollidableEntity;
+import com.github.sef24sp4.common.vector.IVector;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+public class DumbCache implements IPathCaching {
+
+	private final ICollidableEntity entity;
+	private AStar aStarSession;
+	private Map map;
+	private Node entityNode;
+	private Node targetNode;
+	private List<Node> pathList = new LinkedList<>();
+
+	private long lastTimeCalled;
+
+	public DumbCache(final ICollidableEntity entity) {
+		this.entity = entity;
+	}
+
+	@Override
+	public boolean flush() {
+		this.aStarSession = null;
+		return false;
+	}
+
+	@Override
+	public IVector getNextCoordinates(ICollidableEntity collidableEntity, final IVector targetCoordinate) {
```

Comment:
Restructure the method to be on the form
```java
if (!this.isCachedPathValid(targetCoordinates)) {
	/*
	 * Calculate path and assign to pathList
	 * Might be useful to have a method like:
	 */
	this.pathList = this.calculatePath(targetCoordinates);
}

return this.getNextNode().map(Node::getCoordinates).orElse(this.entity.getCoordinates);
```

This avoids duplicate return logic.